### PR TITLE
Add rubocop

### PR DESCRIPTION
**Pull request by benbalter**
Based off of Jekyll's Rubocop config.

### CODE REVIEW

File: lib/jekyll-github-metadata/client.rb [hunk 1]
```diff
@@ -59,9 +59,9 @@ def method_missing(method_name, *args, &block)
 
       def save_from_errors(default = false, &block)
         if block.arity == 1
-          block.call(@client)
+          yield(@client)
         else
-          block.call
+          yield
```
Comment (parkr) [inline lib/jekyll-github-metadata/client.rb]: wont this be a problem? What happens to `block` here? Is yield supposed to know to call block? :confused:

File: script/test [hunk 2]
```diff
@@ -1,3 +1,5 @@
 #!/bin/bash
 echo Testing with $(bundle exec jekyll --version)
 bundle exec rspec --color --require spec_helper $@
+bundle exec rubocop -S -D
+bundle exec gem build jekyll-github-metadata.gemspec
```
Comment (parkr) [inline script/test]: These two actions seem unrelated to rspec. Perhaps they should go in `script/cibuild` instead?

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -149,3 +149,10 @@ def env_args_to_hash(*args)
     Jekyll::GitHubMetadata.logger = Logger.new(StringIO.new) unless ENV["DEBUG"]
   end
 end
+
+def versions
+  require "github-pages"
+  GitHubPages.versions
+rescue LoadError
+  {}
+end
```
Comment (parkr) [inline spec/spec_helper.rb]: Why do this?

**Comment by parkr** [comment]
Thank you! :heart: a couple comments above if you wouldn't mind clarifying.

File: script/test (hunk 2 above)
Comment (benbalter) [inline script/test]: Good call. Will do.

File: spec/spec_helper.rb (hunk 3 above)
Comment (benbalter) [inline spec/spec_helper.rb]: Rubocop complained that it didn't like the original chained call, as it was sandwiched in a lamda inside a hash which ended with `end.call`. This was my best attempt to preserve the original behavior while not angering Rubocop.

Comment (parkr) [inline spec/spec_helper.rb]: Ah ok. We have removed the test with `github-pages` so maybe remove this for now with a comment?

File: lib/jekyll-github-metadata/client.rb (hunk 1 above)
Comment (benbalter) [inline lib/jekyll-github-metadata/client.rb]: That was Rubocop's doing, and tests pass?

Comment (parkr) [inline lib/jekyll-github-metadata/client.rb]: Yeah, it's just confusing 😜 I *think* `yield` doesn't care about the arity of the block so maybe just simplify this to 

```ruby
def save_from_errors(default = false)
  yield @client
rescue # ...
# ...
end
```
